Bug fix for metal-wrapping runes

joshuareagan opened the pull request:
A bug causes this cast to go through:

```
^+(|=(@ 15) ^&(|=(@ 15)))
```

It should nest-fail.  This change makes the cast work correctly with the metal wrapping runes `^&`, `^|`, and `^?`.

joshuareagan commented:
This change corresponds with: https://github.com/urbit/urbit/pull/1044

cgyarvin commented:
That makes total sense — did you change jet too, though?  And does system boot?

Sent from my iPhone

> On Oct 31, 2018, at 12:13 PM, Joshua Reagan <notifications@github.com> wrote:
> 
> @joshuareagan requested your review on: #871 Bug fix for metal-wrapping runes.
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub, or mute the thread.

joshuareagan commented:
Yup, jets changed (there's a PR for that) and it boots and runs fine.  I've done ad hoc testing with casts and everything seems to make sense.

joshuareagan commented:
I should note that I haven't tried booting a ship with the jets off.  Booting without jetted mint seemed a little ambitious.  Should I try that anyway?

joshuareagan commented:
Added the hash (I think I added the correct one...)

The test doesn't seem all that necessary to me -- I can't think of anything that would break the fix other than redoing `mint` altogether, which seems a little out of scope for that kind of testing.

belisarius222 commented:
Yes, please add some tests for this. Shouldn't be too hard.

On Wed, Oct 31, 2018 at 12:37 PM Joe Bryan <notifications@github.com> wrote:

> *@joemfb* requested changes on this pull request.
>
> Could you put the hash of the associated vere commit in
> .travis/pin-vere-commit.txt? That way, the CI will run the appropriate
> vere version.
>
> If you're feeling ambitious, you could add a test in
> /tests/sys/hoon/mint.hoon or somesuch that would fail in the absence of
> this fix, but succeed in the presence of it.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/urbit/arvo/pull/871#pullrequestreview-170432757>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AA1Y9clXlXOnjsKKxFdzXBdLxJW0_SMNks5uqfvegaJpZM4YEwPg>
> .
>
-- 

—
~rovnys-ricfer
https://urbit.org

cgyarvin commented:
Sorry, this response somehow got mis-threaded.  A test is fine!  And this
change also looks fine.

On Wed, Oct 31, 2018 at 1:27 PM Curtis Yarvin <curtis.yarvin@gmail.com>
wrote:

> Don’t try that :-)
>
> Sent from my iPhone
>
> On Oct 31, 2018, at 12:31 PM, Joshua Reagan <notifications@github.com>
> wrote:
>
> I should note that I haven't tried booting a ship with the jets off.
> Booting without jetted mint seemed a little ambitious. Should I try that
> anyway?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/urbit/arvo/pull/871#issuecomment-434815066>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AALyAdS0D10xAjv6iPfIm2gzp2JTraqWks5uqfqIgaJpZM4YEwPg>
> .
>
>
